syntheticEof.ts

```diff
@@ -0,0 +1,83 @@
+import { Transform } from 'node:stream';
+
+// keep in sync with lib.rs::SYNTHETIC_EOF
+export const SYNTHETIC_EOF = Buffer.from('\x1B]7878\x1B\\');
+
+function getCommonPrefixLength(buffer: Buffer) {
+  for (let prefixLen = SYNTHETIC_EOF.length; prefixLen >= 1; prefixLen--) {
+    const suffix = buffer.subarray(buffer.length - prefixLen);
+    const prefix = SYNTHETIC_EOF.subarray(0, prefixLen);
+
+    if (suffix.equals(prefix)) {
+      return prefixLen;
+    }
+  }
+  return 0;
+}
+
+export class SyntheticEOFDetector extends Transform {
+  buffer: Buffer;
+  maxBufferSize: number;
+
+  constructor(options = {}) {
+    super(options);
+    this.buffer = Buffer.alloc(0);
+    this.maxBufferSize = SYNTHETIC_EOF.length - 1;
+  }
+
+  _transform(chunk: Buffer, encoding: string, callback: () => void) {
+    // combine any leftover buffer with new chunk
+    // and look for synthetic EOF in the combined data
+    const searchData = Buffer.concat([this.buffer, chunk]);
+    const eofIndex = searchData.indexOf(SYNTHETIC_EOF);
+
+    // found EOF - emit data before it
+    if (eofIndex !== -1) {
+      const beforeEOF = searchData.subarray(0, eofIndex);
+      const afterEOF = searchData.subarray(eofIndex + SYNTHETIC_EOF.length);
+
+      if (beforeEOF.length > 0) {
+        this.push(Buffer.from(beforeEOF));
+      }
+
+      this.emit('synthetic-eof');
+
+      // Continue processing remaining data (might have more EOFs)
+      if (afterEOF.length > 0) {
+        this._transform(Buffer.from(afterEOF), encoding, callback);
+        return;
+      }
```

Comment:
should we move to `while (true)` instead so that we don't blow up the stack in degenerate cases? Also saves us some reallocation of `searchData`